src/pkg/packager/publish.go

```diff
@@ -62,14 +40,32 @@ func parseZarfLayerMediaType(filename string) string {
 // Authentication is handled via the Docker config file created w/ `zarf tools registry login`
 func (p *Packager) Publish() error {
 	p.cfg.DeployOpts.PackagePath = p.cfg.PublishOpts.PackagePath
+	var ref registry.Reference
+	if utils.IsDir(p.cfg.PublishOpts.PackagePath) {
+		err := p.loadSkeleton()
+		if err != nil {
+			return err
+		}
 
-	// Extract the first layer of the tarball
-	if err := archiver.Unarchive(p.cfg.DeployOpts.PackagePath, p.tmp.Base); err != nil {
-		return fmt.Errorf("unable to extract the package: %w", err)
-	}
+		ref, err = p.ref("skeleton")
+		if err != nil {
+			return err
+		}
+	} else {
+		// Extract the first layer of the tarball
+		if err := archiver.Unarchive(p.cfg.DeployOpts.PackagePath, p.tmp.Base); err != nil {
+			return fmt.Errorf("unable to extract the package: %w", err)
+		}
+
+		err := p.readYaml(p.tmp.ZarfYaml, true)
+		if err != nil {
+			return fmt.Errorf("unable to read the zarf.yaml in %s: %w", p.tmp.Base, err)
+		}
 
-	if err := p.readYaml(p.tmp.ZarfYaml, true); err != nil {
-		return fmt.Errorf("unable to read the zarf.yaml in %s: %w", p.tmp.Base, err)
+		ref, err = p.ref("")
```

Comment:
Nit; can we clean this up a bit?

I would recommend creating a variable for the suffix at the beginning of this function, populate the `skeleton` as needed in the `if` block. and then call `p.ref(refSuffix)` outside of this `else` block.